src/14/HW/SearchableTodoList.ts

```diff
@@ -0,0 +1,14 @@
+import { Note } from "./Note";
+import { TodoList } from "./TodoList";
+
+export class SearchableTodoList extends TodoList {
+    public searchNotes(keyword: string, searchBy: 'title' | 'content'): Note[] {
+        if (searchBy === 'title') {
+            return this.getAllNotes().filter(note => note.title.includes(keyword));
+        } else if (searchBy === 'content') {
+            return this.getAllNotes().filter(note => note.content.includes(keyword));
+        } else {
+            throw new Error('Invalid search criteria. Use "title" or "content".');
```

Comment:
This implementation has limitations. It requires adding a second parameter, even when it’s unnecessary. A better approach would be to divide this logic into two methods or, at the very least, assign a default value to the second parameter.